Scripts/RemediationScripts/Remediate-AnonymousAccessOnContainers.ps1

```diff
@@ -195,63 +256,39 @@ function Remove-AnonymousAccessOnContainers
     Write-Host "Starting with subscription [$($SubscriptionId)]..."
 
     Write-Host "`n"
-    Write-Host "*** To perform remediation for disabling anonymous access on containers user must have atleast contributor access on storage account(s) of subscription: [$($SubscriptionId)] ***" -ForegroundColor $([Constants]::MessageType.Warning)
-    Write-Host "`n" 
+    Write-Host "*** WARNING: To perform remediation for disabling anonymous access on containers user must have atleast contributor access on Storage account(s) of subscription: [$($SubscriptionId)] ***" -ForegroundColor $([Constants]::MessageType.Warning)
+    Write-Host $([Constants]::DoubleDashLine) 
 
-    Write-Host "Validating whether the current user [$($currentSub.Account.Id)] has valid account type [User] to run the script for subscription [$($SubscriptionId)]..."
-    
-    # Safe Check: Checking whether the current account is of type User
-    if($currentSub.Account.Type -ne "User")
-    {
-        Write-Host "Warning: This script can only be run by user account type." -ForegroundColor $([Constants]::MessageType.Warning)
-        break;
-    }
-    
-    # Safe Check: Current user must have Owner/Contributor/User Access Administrator access over the subscription.
-    $currentLoginRoleAssignments = Get-AzRoleAssignment -SignInName $currentSub.Account.Id -Scope "/subscriptions/$($SubscriptionId)";
-
-    $requiredRoleDefinitionName = @("Owner", "Contributor", "User Access Administrator")
-    if(($currentLoginRoleAssignments | Where { $_.RoleDefinitionName -in $requiredRoleDefinitionName} | Measure-Object).Count -le 0 )
-    {
-        Write-Host "Warning: This script can only be run by an [$($requiredRoleDefinitionName -join ", ")]." -ForegroundColor Yellow
-        return;
-    }
-    else
-    {
-        Write-Host "Current user [$($currentSub.Account.Id)] has the required permission for subscription [$($SubscriptionId)]." -ForegroundColor Green
-    }
-
-    Write-Host "Validation succeeded." -ForegroundColor $([Constants]::MessageType.Update)
-    Write-Host "`n"
-    Write-Host "Fetching storage account(s)..."
+    Write-Host "Fetching Storage account(s)..."
     
     # Array to store resource context
     $resourceContext = @()
     $controlIds = "Azure_Storage_AuthN_Dont_Allow_Anonymous"
     
-    # If json path not given fetch all storage account.
-    if([string]::IsNullOrWhiteSpace($Path))
+    # If json path not given fetch all Storage account.
+    if([string]::IsNullOrWhiteSpace($Path) -or [string]::IsNullOrWhiteSpace($FilePath))
```

Comment:
If anyone filePath is empty it will list all storage account

Reply:
Fixed the issue.